src/app/fdctl/run/tiles/fd_sender.c

```diff
@@ -314,63 +304,38 @@ after_frag( void *             _ctx,
-      }
-
-      fd_voter_t voter = {
-        .vote_acct_addr              = ctx->vote_acct_addr,
-        .vote_authority_pubkey       = ctx->identity_key,
-        .validator_identity_pubkey   = ctx->identity_key,
-        .voter_sign_arg              = ctx,
-        .vote_authority_sign_fun     = vote_txn_signer,
-        .validator_identity_sign_fun = vote_txn_signer
-      };
-
-      uchar txn_meta_buf[ FD_TXN_MAX_SZ ];
-      uchar * msg_to_gossip = fd_chunk_to_laddr( ctx->gossip_out_mem, ctx->gossip_out_chunk );
-      ulong vote_txn_sz = fd_vote_txn_generate( &voter,
-                                                &compate_vote_state_update,
-                                                recent_blockhash.uc,
-                                                txn_meta_buf,
-                                                msg_to_gossip );
-      /* send to leader */
-      fd_shred_dest_weighted_t * leader_dest = NULL;
-      int res = get_current_leader_tpu_vote_contact( ctx, &leader_dest );
-      if( res==0 ) {
-        send_packet( ctx, leader_dest->ip4, leader_dest->port, msg_to_gossip, vote_txn_sz, 0UL );
-      }
-
-      /* send to gossip */
-      fd_mcache_publish( ctx->gossip_out_mcache, ctx->gossip_out_depth, ctx->gossip_out_seq, 1UL, ctx->gossip_out_chunk,
-        vote_txn_sz, 0UL, 0, 0 );
-      ctx->gossip_out_seq   = fd_seq_inc( ctx->gossip_out_seq, 1UL );
-      ctx->gossip_out_chunk = fd_dcache_compact_next( ctx->gossip_out_chunk, vote_txn_sz,
-                                                      ctx->gossip_out_chunk0, ctx->gossip_out_wmark );
-      /* send to pack */
-      uchar * msg_to_pack = fd_chunk_to_laddr( ctx->pack_out_mem, ctx->pack_out_chunk );
-      memcpy( msg_to_pack, msg_to_gossip, vote_txn_sz );
-      fd_mcache_publish( ctx->pack_out_mcache, ctx->pack_out_depth, ctx->pack_out_seq, 1UL, ctx->pack_out_chunk,
-        vote_txn_sz, 0UL, 0, 0 );
-      ctx->pack_out_seq    = fd_seq_inc( ctx->pack_out_seq, 1UL );
-      ctx->pack_out_chunk = fd_dcache_compact_next( ctx->pack_out_chunk, vote_txn_sz, ctx->pack_out_chunk0,
-          ctx->pack_out_wmark );
-    } FD_SCRATCH_SCOPE_END;
+    fd_txn_p_t * txn = (fd_txn_p_t *)fd_type_pun(ctx->txn_buf);
+    
+    /* sign the txn */
+    uchar * signature = txn->payload + TXN(txn)->signature_off;
+    uchar * message   = txn->payload + TXN(txn)->message_off;
+    ulong message_sz  = txn->payload_sz - TXN(txn)->message_off;
+    fd_keyguard_client_sign( ctx->keyguard_client, signature, message, message_sz );
+
+    uchar * msg_to_gossip = fd_chunk_to_laddr( ctx->gossip_out_mem, ctx->gossip_out_chunk );
+    memcpy( msg_to_gossip, txn->payload, txn->payload_sz );
+
+    /* send to leader */
+    fd_shred_dest_weighted_t * leader_dest = NULL;
+    int res = get_current_leader_tpu_vote_contact( ctx, &leader_dest );
+    if( res==0 ) {
+      send_packet( ctx, leader_dest->ip4, leader_dest->port, msg_to_gossip, txn->payload_sz, 0UL );
+    }
```

Comment:
log warning?

Reply:
it would could be too noisy on start up. would rather just have metrics for it. will add a todo